### DISCUSSION TITLE
Using brightness sensor from headunit as filter

### DISCUSSION THREAD

**Discussion by Tronje-the-Falconer** [maca88/SmartBikeLights#73] [General]
Such a great and useful app, I love it!

Is there a way to use the head unit's brightness sensor to control the lights? That would be great.
It could automatically turn on the lights when you enter a tunnel, or turn them on when it gets dark in the evening.

**Comment by maca88** [comment]
Hi,

Unfortunately Garmin does not provide an API for the brightness sensor, here you can find the feature request:
https://forums.garmin.com/developer/connect-iq/i/bug-reports/ambient-light-sensor-api-access

If you own a solar device then it is possible to use the "Solar intensity" filter which works similar as the brightness/ambient sensor.
If not, then you can use the "GPS accuracy < Good" filter for the tunnel and the "Timespan" filter with the "sunset/sunrise" option for turning the lights on in the evening.

**Reply by frt** [reply]
Do you know some approximate values of "Solar intensity" for some situations, such as night with urban ilumination,  roda without ilumination, sunny day, cloudy day, fog, etc.? That would help a lot in using this data in the configuration.
